### PR TITLE
NIFI-11603: Removed NetworkUtils.getAvailableUdpPort, NetworkUtils.ge…

### PR DESCRIPTION
**Pull request by markap14**
…tAvailableTcpPort and NetworkUtils.getAvailablePort as well as NetworkUtils.isListening

# Summary

[NIFI-00000](https://issues.apache.org/jira/browse/NIFI-00000)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by joewitt** [comment]
building on a local osx and linux set of machines now.

**Comment by joewitt** [comment]
with -Pinclude-grpc it appears some bits were missed


[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:testCompile (default-testCompile) on project nifi-grpc-processors: Compilation failure: Compilation failure:
[ERROR] /development/code/nifi.git/nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-processors/src/test/java/org/apache/nifi/processors/grpc/TestListenGRPC.java:[78,37] error: cannot find symbol
[ERROR]   symbol:   method getAvailableTcpPort()
[ERROR]   location: class NetworkUtils
[ERROR] /development/code/nifi.git/nifi-nar-bundles/nifi-grpc-bundle/nifi-grpc-processors/src/test/java/org/apache/nifi/processors/grpc/TestListenGRPC.java:[90,37] error: cannot find symbol
[ERROR]   symbol:   method getAvailableTcpPort()
[ERROR]   location: class NetworkUtils
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :nifi-grpc-processors

**Comment by markap14** [comment]
Thanks @joewitt missed that one since I didn't build with the profile

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestListenUDP.java [hunk 1]
```diff
@@ -173,7 +170,7 @@ public void testRunWhenNoEventsAvailable() {
 
     @Test
     public void testWithSendingHostAndPortSameAsSender() throws IOException, InterruptedException {
-        final Integer sendingPort = NetworkUtils.getAvailableUdpPort();
+        final int sendingPort = 27911;
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestListenUDP.java:173]: Will this hard-coded port result in intermittent failures? Can it be set to `0` for constructing the `DatagramSocket`?

**Review by exceptionfactory — COMMENTED** [review]
Thanks for addressing these issues @markap14, the approach looks good. I only noticed two minor things, one was an unused member variable and the other was a hard-coded client port number, otherwise this looks good.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestListenUDP.java (hunk 1 above)
Comment (markap14) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestListenUDP.java:173]: Good catch, I meant to come back to that one, as I wasn't 100% sure what was happening there. Will address, though.

**Pull request closed without merging**